### PR TITLE
Hugging Face Language Models Implementation

### LINKED ISSUE #2336: Implement HuggingFace Language Modeling Estimators
**Is your feature request related to a problem? Please describe.**
The next step of integrating HuggingFace into ART is to add support for the language modeling estimators. This involves creating ART estimator wrappers for the HuggingFace text models. These estimators should support

**Describe the solution you'd like**
A new module will be created: `art.estimators.language_modeling` which will be where all of the new HuggingFace language modeling estimators will be implemented.

~A new estimator will be created for each language modeling task (e.g., masked LM, sequence classification, next sentence prediction, etc.). Each estimator will be named accordingly (e.g., `HuggingFaceMaskedLM`, `HuggingFaceSequenceClassificationLM`, `HuggingFaceNextSequencePredictionLM`, etc.). This is due to the fact that the expected input and output for each task is unique.~

Each estimator will take in a HuggingFace model and the corresponding tokenizer. In this approach, the model and tokenizer will be coupled in the same wrapper. This is the simplest approach since the tokenizer is specific to the text model and is not very useful standalone for ART's use cases.

**Describe alternatives you've considered**
The tokenizer can be made its own standalone module that is passed in to the ART wrapper. However, the tokenizer by itself is not very useful since it is dependent on the model (BERT, GPT-2, T5, etc.) and adds unnecessary complexity to creating the language model. If needed, the tokenizer can always be decoupled from the model and made standalone at a later point.

**Additional context**
The naming for the module and estimators are not finalized and are open to suggestions.

### PR DESCRIPTION
**Pull request by f4str**
# Description

Implementation of language models under a new `art.estimators.language_modeling` submodule. Currently only Hugging Face language models using a PyTorch back-end have been implemented. This is implemented as the `HuggingFaceLanguageModel` which is a generic estimator that is able to run basic functionality on any Hugging Face model.

Fixes #2336 

## Type of change

Please check all relevant options.

- [ ] Improvement (non-breaking)
- [ ] Bug fix (non-breaking)
- [x] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# Testing

Please describe the tests that you ran to verify your changes. Consider listing any relevant details of your test configuration.

- [ ] Test A
- [ ] Test B

**Test Configuration**:
- OS
- Python version
- ART version or commit number
- TensorFlow / Keras / PyTorch / MXNet version

# Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] My changes have been tested using both CPU and GPU devices

### CODE REVIEW

**Comment by jetlime** [comment]
Hey @f4str , what's the status on this PR? Shall it be picked up?
